system_modules/napsequence/src/sequencetracksegmentcurve.cpp

```diff
@@ -4,6 +4,11 @@
 
 #include "sequencetracksegmentcurve.h"
 
+RTTI_BEGIN_CLASS_NO_DEFAULT_CONSTRUCTOR(nap::SequenceTrackSegmentCurveBase)
```

Comment:
If this is not a resource consider defining it as a `STRUCT`

Reply:
It's a resource (purely used by the SequencePlayer, but the sequencer treats it as a resource and serialize/deserializes this)

Reply:
A type doesn't have to be a `nap::Object` to be serializable. If it's a segment that's embedded and local to another object it could be better to treat it as a copy constructable struct instead of object. Why does it need to be an object with a unique ID?

Reply:
agree, also see https://github.com/napframework/nap/pull/29#discussion_r1736079715